frontend/components/course/Tile.tsx

```diff
@@ -2,17 +2,84 @@ import Image from "next/image";
 import Link from "next/link";
 import { FC } from "react";
 
+import { EnrollmentStatus_enum } from "../../__generated__/globalTypes";
+import { enrollmentStatusForCourse } from "../../helpers/courseHelpers";
 import { CourseList_Course } from "../../queries/__generated__/CourseList";
+import { CourseListWithEnrollments_Course } from "../../queries/__generated__/CourseListWithEnrollments";
 
 interface IProps {
-  course: CourseList_Course;
+  course: CourseList_Course | CourseListWithEnrollments_Course;
 }
 
+const colorForEnrollmentStatus = (
+  status: EnrollmentStatus_enum | "NOT_APPLIED"
+): string => {
+  if (
+    status === EnrollmentStatus_enum.APPLIED ||
+    status === EnrollmentStatus_enum.CONFIRMED
+  ) {
+    return "bg-edu-course-current";
+  }
+  if (
+    status === EnrollmentStatus_enum.REJECTED ||
+    status === EnrollmentStatus_enum.ABORTED
+  ) {
+    return "bg-gray-300";
+  }
+  if (status === EnrollmentStatus_enum.INVITED) {
+    return "bg-edu-course-invited";
+  }
+  return "";
+};
+
+const CourseStatusIndicator: FC<{
+  enrollmentStatus: EnrollmentStatus_enum | "NOT_APPLIED";
+}> = ({ enrollmentStatus }) => {
+  const color = colorForEnrollmentStatus(enrollmentStatus);
+
+  if (enrollmentStatus === EnrollmentStatus_enum.COMPLETED) {
+    return (
+      <div className="absolute top-0 right-3">
+        <Image src="/images/course/completed_flag.svg" width={37} height={46} />
+      </div>
+    );
+  }
+  // if (
+  //   enrollmentStatus === EnrollmentStatus_enum.APPLIED ||
+  //   enrollmentStatus === EnrollmentStatus_enum.CONFIRMED ||
+  //   enrollmentStatus === EnrollmentStatus_enum.REJECTED ||
+  //   enrollmentStatus === EnrollmentStatus_enum.ABORTED ||
+  //   enrollmentStatus === EnrollmentStatus_enum.INVITED
+  // )
+  else {
+    return (
+      <div className="absolute top-3 right-3">
+        <div className={`rounded-full w-9 h-9 ${color}`} />;
+      </div>
+    );
+  }
+};
+
 export const Tile: FC<IProps> = ({ course }) => {
+  const enrollmentStatus = enrollmentStatusForCourse(course);
+
+  const semester = course.Semester;
+
+  const currentDate = new Date();
+  const isCurrentSemester =
+    !!semester &&
+    semester?.Start <= currentDate &&
```

Comment:
According to https://github.com/edu-hub-project/edu-hub/issues/35#issuecomment-882622783 this should be:
```suggestion
    semester?.ApplicationStart <= currentDate &&
```

Reply:
Good catch. Fixed that.